tests/test_http_status.py

```
@@ -0,0 +1,155 @@
+# ---------------------------------------------------------------------------
+# Integration: fetch_lever_jobs
+# ---------------------------------------------------------------------------
+
+class TestLeverHttpStatus:
+    """Verify fetch_lever_jobs uses is_retryable_status correctly."""
+
+    def test_404_stops_immediately(self, monkeypatch):
+        """404 is non-retryable — should return [] without retrying."""
+        call_count = 0
+
+        def fake_limited_get(url, **kwargs):
```

Ruff (0.15.6)

[warning] Missing return type annotation for private function `fake_limited_get`

Add return type annotation: `Never`

(ANN202)